front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java

```diff
@@ -52,17 +53,18 @@
 @ConditionalOnExpression("${spinnaker.gcs.enabled:false}")
 @EnableConfigurationProperties(GcsProperties.class)
 public class GcsConfig {
-  // Refresh every 10 minutes. In practice this either doesnt matter because refreshes are fast enough,
-  // or should be finer tuned. But it seems silly to refresh at a fast rate when changes are generally infrequent.
-  // Actual queries always check to see if the cache is out of date anyway. So this is mostly for the benefit of
-  // keeping other replicas up to date so that last-minute updates have fewer changes in them.
+  @Autowired
+  Registry registry;
+
   private final Logger log = LoggerFactory.getLogger(getClass());
-  private static int APPLICATION_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
-  private static int PROJECT_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
-  private static int NOTIFICATION_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
-  private static int PIPELINE_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
-  private static int PIPELINE_STRATEGY_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
-  private static int SERVICE_ACCOUNT_REFRESH_MS = (int) TimeUnit.MINUTES.toMillis(1);
+  private static int APPLICATION_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(15);
+  private static int APPLICATION_PERMISSIONS_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(45);
+  private static int PROJECT_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(30);
+  private static int NOTIFICATION_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(30);
+  private static int PIPELINE_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(10);
+  private static int PIPELINE_STRATEGY_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(20);
+  private static int SERVICE_ACCOUNT_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(30);
+  private static int SNAPSHOT_REFRESH_MS = (int) TimeUnit.SECONDS.toMillis(60);
```

Comment:
If we're trying to stay in sync with S3, these values should probably be constants in a common module.